### DISCUSSION TITLE
Update seems to get stuck every now and then.

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#394] [General]
I have just gotten my Octopus Mini, But I have noticed that the sensors get stuck now and then, is any one else noticing this?
<img width="614" alt="Screenshot 2023-09-01 at 11 39 38" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/6abdeca4-6de8-4c42-873f-28852ed8eed7">

as you can see it says updated 10 mins ago, and this then causes the data to be fixed at that point until it gets new data.

**Comment by BottlecapDave** [comment]
Hello and sorry that you're having issues. There are a number of reasons why updates can be delayed

1. The OE API is being flakey. I've seen numerous times the APIs reporting server errors, which will cause an update to fail. These should be present in your HA logs if occuring if your debug levels are set to `sebug`.
2. You're hitting the API rate limits. This can occur of your Home Mini is also reporting gas updates, as there's a rate limit of 100 calls per 1 hour. You can adjust this time by configuring the integration and adjusting the update times (which defaults to once a minute).

**Comment by andyblac** [comment]
I thought the gas only updates once every 30 mins ?, at least the OE app has no live data for GAS, and tell me to use day view. Maybe it was because it was new and i was playing with the OE app, and your integration was requesting API call too and thus hitting the limit. I do recall the OE app had a few "oops, something went wrong" message as well.

**Reply by BottlecapDave** [reply]
The gas updates less frequently to the electricity (therefore no "live" view), but the integration checks for data as frequently (defaults once a miniute) as there's no set time (e.g. on the hour) to know when the data is available. But yeah, if you're using the integration and have gas you'll need to increase the time between updates in order to not hit limits. You might need to increase further if you use the app frequently too. There is an [active discussion](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/255) where someone is trying to get the 100 API calls an hour increased, but there's minimal interaction with OE.
